Wikipedia/Code/ArticleViewController+Announcements.swift

```diff
@@ -4,22 +4,35 @@ import CocoaLumberjackSwift
 extension ArticleViewController {
     
     func showAnnouncementIfNeeded() {
-        guard UserDefaults.standard.shouldCheckForArticleAnnouncements,
-              ((isInValidSurveyCampaignAndArticleList && userHasSeenSurveyPrompt) || !isInValidSurveyCampaignAndArticleList) else {
+        guard ((isInValidSurveyCampaignAndArticleList && userHasSeenSurveyPrompt) || !isInValidSurveyCampaignAndArticleList) else {
             return
         }
-        let predicate = NSPredicate(format: "placement == 'article'")
-        let contentGroup = dataStore.viewContext.newestVisibleGroup(of: .announcement, with: predicate)
+        let predicate = NSPredicate(format: "placement == 'article' && isVisible == YES")
+        let contentGroups = dataStore.viewContext.orderedGroups(of: .announcement, with: predicate)
+        
+        //get the first content group with a valid date
+        let contentGroup = contentGroups?.first(where: { (group) -> Bool in
+            let currentDate = Date()
```

Comment:
Move this a couple lines up for performance?

Reply:
Good call, this is done!